### PR TITLE
Abstract Factory Initial Commit

### PR DESCRIPTION
**Pull request by Samuelpf98**
- Added abstract factory for movies
- fixed add and remove movies
- still needs more cleaning up, just initial commit

### CODE REVIEW

File: logic_pkg/edit_screenings_logic.py [hunk 1]
```diff
@@ -6,47 +6,30 @@
...
-        text = sg.popup_get_text("Add screening in format 'MovieTitle,Screen,Time1,...,TimeN")
-        if text != None:
-            v = window['-MOVIES-'].get_list_values()
-            m = utils.convertToDisplayForm(text)
-            if m.endswith(":  ") != True:
-                v.append(m)
-                window['-MOVIES-'].update(values=v)
+        text = sg.popup_get_text("Add screening in format 'MovieTitle, Screen No, 2D or 3D, Subtitled or Not Subtitled, Time1,...,TimeN,'")
+        warning_text = "Screenings must be in format ''MovieTitle, Screen No, 2D or 3D, Subtitled or Not Subtitled, Time1,...,TimeN,'"
+        if text != "":
```
Comment (Lennaayy) [inline logic_pkg/edit_screenings_logic.py]: Add in Nonetype here, as it causes a crash when you press X

File: logic_pkg/edit_screenings_logic.py [hunk 2]
```diff
@@ -6,47 +6,30 @@
...
     if event == 'Add Screening':
-        text = sg.popup_get_text("Add screening in format 'MovieTitle,Screen,Time1,...,TimeN")
-        if text != None:
-            v = window['-MOVIES-'].get_list_values()
-            m = utils.convertToDisplayForm(text)
-            if m.endswith(":  ") != True:
-                v.append(m)
-                window['-MOVIES-'].update(values=v)
+        text = sg.popup_get_text("Add screening in format 'MovieTitle, Screen No, 2D or 3D, Subtitled or Not Subtitled, Time1,...,TimeN,'")
+        warning_text = "Screenings must be in format ''MovieTitle, Screen No, 2D or 3D, Subtitled or Not Subtitled, Time1,...,TimeN,'"
```
Comment (Lennaayy) [inline logic_pkg/edit_screenings_logic.py]: Spacing in these doesn't match the input. When using space after every comma it adds the last movie in the DB twice instead, either strip() the input or change the text maybe?

**Review by Lennaayy — CHANGES_REQUESTED** [review]
Minor comments in review to be addressed.

Also noticed you can still redeem a booked ticket for a movie that has been deleted from the DB, should this be added as a bug or are we not going to address it?

File: logic_pkg/edit_screenings_logic.py (hunk 2 above)
Comment (Samuelpf98) [reply logic_pkg/edit_screenings_logic.py]: I mentioned in discord to just not add spaces for the current code. For now I'll just change the text to reflect this

File: logic_pkg/edit_screenings_logic.py (hunk 1 above)
Comment (Samuelpf98) [reply logic_pkg/edit_screenings_logic.py]: Didn't try this use case, will fix 👍

**Review by Lennaayy — APPROVED** [review]
Looks Great To Me Buckaroo

**Review by LukeMaher98 — APPROVED** [review]
Everything appears in order, looks good

**Pull request merged**